### PR TITLE
test: regress: Use Minisat for tests that are not supported with CaDiCaL.

### PR DESCRIPTION
**Pull request by aniemetz**

### CODE REVIEW

File: src/smt/set_defaults.cpp [hunk 1]
```diff
@@ -192,12 +201,26 @@ void SetDefaults::setDefaultsPre(Options& opts)
...
+      ss << reasonNoProofs.str() << " not supported with proofs or unsat cores";
+      throw OptionException(ss.str());
+    }
     SET_AND_NOTIFY(smt, produceProofs, true, "option requiring proofs");
   }
 
   // this check assumes the user has requested *full* proofs
   if (opts.smt.produceProofs)
   {
+    if (opts.prop.satSolver == options::SatSolverMode::CADICAL)
```
Comment (ajreynol) [inline src/smt/set_defaults.cpp:217]: Shouldn't this check be subsumed by the one in `incompatibleWithProofs`?

File: src/smt/set_defaults.cpp [hunk 2]
```diff
@@ -335,6 +358,15 @@ void SetDefaults::setDefaultsPre(Options& opts)
                            sygusInference,
                            options::SygusInferenceMode::OFF,
                            "internal subsolver");
+    // proofs/unsat cores not yet supported with CaDiCaL
```
Comment (ajreynol) [inline src/smt/set_defaults.cpp:361]: I would expect this to be unecessary now with the other changes.

File: test/regress/cli/regress0/arith/issue5219-conflict-rewrite.smt2 [hunk 3]
```diff
@@ -2,7 +2,6 @@
 ; COMMAND-LINE: --theoryof-mode=term --nl-icp
 ; EXPECT: sat
 (set-logic QF_NRA)
-(set-option :produce-proofs true)
```
Comment (ajreynol) [inline test/regress/cli/regress0/arith/issue5219-conflict-rewrite.smt2:5]: I think we should keep `produce-proofs` set for benchmarks that are not "unsat", as this regression otherwise does not test proofs.  I believe benchmarks like these were added since a "sat" benchmark had an error e.g. in bookkeeping proofs.

File: test/regress/cli/regress0/nl/proj-issue-465-asan-proofs.smt2 [hunk 4]
```diff
@@ -1,4 +1,4 @@
-; COMMAND-LINE: -i -q --produce-proofs
+; COMMAND-LINE: -i -q
```
Comment (ajreynol) [inline test/regress/cli/regress0/nl/proj-issue-465-asan-proofs.smt2:1]: Ditto.

File: test/regress/cli/regress0/preprocess/proj-issue304-circuit-prop-xor.smt2 [hunk 5]
```diff
@@ -1,6 +1,5 @@
 ; EXPECT: sat
 (set-logic ALL)
-(set-option :produce-proofs true)
```
Comment (ajreynol) [inline test/regress/cli/regress0/preprocess/proj-issue304-circuit-prop-xor.smt2:3]: Ditto.

File: test/regress/cli/regress0/preprocess/proj-issue305-circuit-prop-ite-a.smt2 [hunk 6]
```diff
@@ -1,10 +1,9 @@
 ; EXPECT: sat
 (set-logic ALL)
-(set-option :produce-proofs true)
```
Comment (ajreynol) [inline test/regress/cli/regress0/preprocess/proj-issue305-circuit-prop-ite-a.smt2:3]: Ditto.

File: test/regress/cli/regress0/preprocess/proj-issue305-circuit-prop-ite-b.smt2 [hunk 7]
```diff
@@ -1,10 +1,9 @@
 ; EXPECT: sat
 (set-logic ALL)
-(set-option :produce-proofs true)
```
Comment (ajreynol) [inline test/regress/cli/regress0/preprocess/proj-issue305-circuit-prop-ite-b.smt2:3]: Ditto.

File: test/regress/cli/regress0/preprocess/proj-issue305-circuit-prop-ite-c.smt2 [hunk 8]
```diff
@@ -1,10 +1,9 @@
 ; EXPECT: sat
 (set-logic ALL)
-(set-option :produce-proofs true)
```
Comment (ajreynol) [inline test/regress/cli/regress0/preprocess/proj-issue305-circuit-prop-ite-c.smt2:3]: Ditto.

File: test/regress/cli/regress0/preprocess/proj-issue305-circuit-prop-ite-d.smt2 [hunk 9]
```diff
@@ -1,10 +1,9 @@
 ; EXPECT: sat
 (set-logic ALL)
-(set-option :produce-proofs true)
```
Comment (ajreynol) [inline test/regress/cli/regress0/preprocess/proj-issue305-circuit-prop-ite-d.smt2:3]: Ditto.

File: test/regress/cli/regress0/preprocess/proj-issue309-circuit-prop-ite.smt2 [hunk 10]
```diff
@@ -1,6 +1,5 @@
 ; EXPECT: sat
 (set-logic ALL)
-(set-option :produce-proofs true)
```
Comment (ajreynol) [inline test/regress/cli/regress0/preprocess/proj-issue309-circuit-prop-ite.smt2:3]: Ditto.

File: test/regress/cli/regress0/preprocess/proj-issue332-circuit-prop-xor.smt2 [hunk 11]
```diff
@@ -1,6 +1,5 @@
 ; EXPECT: sat
 (set-logic ALL)
-(set-option :produce-proofs true)
```
Comment (ajreynol) [inline test/regress/cli/regress0/preprocess/proj-issue332-circuit-prop-xor.smt2:3]: Ditto.

File: test/regress/cli/regress0/proofs/issue9669-clause-level-opt-incremental.smt2 [hunk 12]
```diff
@@ -1,4 +1,4 @@
-; COMMAND-LINE: -i --check-proofs
```
Comment (ajreynol) [inline test/regress/cli/regress0/proofs/issue9669-clause-level-opt-incremental.smt2:1]: Ditto, but change to `--produce-proofs`.

File: test/regress/cli/regress0/proofs/proj-issue326-nl-bounds-check.smt2 [hunk 13]
```diff
@@ -1,6 +1,5 @@
 ; EXPECT: sat
 (set-logic ALL)
-(set-option :produce-proofs true)
```
Comment (ajreynol) [inline test/regress/cli/regress0/proofs/proj-issue326-nl-bounds-check.smt2:3]: Ditto.

File: test/regress/cli/regress0/proofs/proj-issue492-skolem-lemma-pf.smt2 [hunk 14]
```diff
@@ -1,5 +1,4 @@
 (set-logic ALL)
 (set-info :status sat)
-(set-option :produce-proofs true)
```
Comment (ajreynol) [inline test/regress/cli/regress0/proofs/proj-issue492-skolem-lemma-pf.smt2:3]: Ditto.

File: test/regress/cli/regress0/proofs/project-issue330-eqproof.smt2 [hunk 15]
```diff
@@ -2,6 +2,5 @@
 (set-logic QF_SLIA)
 (declare-const x String)
 (declare-const x1 Int)
-(set-option :produce-proofs true)
```
Comment (ajreynol) [inline test/regress/cli/regress0/proofs/project-issue330-eqproof.smt2:5]: Ditto.

File: test/regress/cli/regress1/nl/factor_agg_s.smt2 [hunk 16]
```diff
@@ -3,7 +3,6 @@
 ; EXPECT: sat
 (set-logic QF_NRA)
 (set-info :status sat)
-(set-option :produce-proofs true)
```
Comment (ajreynol) [inline test/regress/cli/regress1/nl/factor_agg_s.smt2:6]: Ditto.

File: test/regress/cli/regress1/proofs/eq-engine-corner-case-constants-disequal.smt2 [hunk 17]
```diff
@@ -1,4 +1,3 @@
-; COMMAND-LINE: --produce-proofs
```
Comment (ajreynol) [inline test/regress/cli/regress1/proofs/eq-engine-corner-case-constants-disequal.smt2:1]: Ditto.

File: test/regress/cli/regress1/quantifiers/recfact.cvc.smt2 [hunk 18]
```diff
@@ -1,8 +1,5 @@
 ; EXPECT: unsat
-; COMMAND-LINE:
-; COMMAND-LINE: --produce-proofs
-;; The second command line option, other than the default, is to test
-;; unsat core checking with proofs, which at one point had issues for
+;; Unsat core checking with proofsh at one point had issues for
```
Comment (ajreynol) [inline test/regress/cli/regress1/quantifiers/recfact.cvc.smt2:2]:
```suggestion
;; Unsat core checking with proofs at one point had issues for
```

File: test/regress/cli/regress1/sets/issue10941-open-pf.smt2 [hunk 19]
```diff
@@ -1,4 +1,4 @@
-; COMMAND-LINE: --produce-proofs -q
```
Comment (ajreynol) [inline test/regress/cli/regress1/sets/issue10941-open-pf.smt2:1]: Ditto.

**Review by ajreynol — CHANGES_REQUESTED** [review]

File: src/smt/set_defaults.cpp (hunk 1 above)
Comment (aniemetz) [reply src/smt/set_defaults.cpp:217]: It is not, for some reason.

File: src/smt/set_defaults.cpp (hunk 2 above)
Comment (aniemetz) [reply src/smt/set_defaults.cpp:361]: indeed

Comment (ajreynol) [reply src/smt/set_defaults.cpp:361]: SG, can you delete the commented out code now?

**Review by ajreynol — APPROVED** [review]
